### PR TITLE
feat: Require internet permission

### PR DESCRIPTION
**Pull request by bruno-garcia**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Won't require user configuration in their app to add the permissions required by Sentry's SDK.

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] All tests passing

### CODE REVIEW

File: sentry-android-core/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -9,4 +9,5 @@
       android:exported="false"
       android:initOrder="2147483647" />
   </application>
+  <uses-permission android:name="android.permission.INTERNET" />
```
Comment (ninniuz) [inline sentry-android-core/src/main/AndroidManifest.xml]: I don't know why but you are asking apps to also include `android.permission.ACCESS_NETWORK_STATE` in the [docs](http://tools.android.com/tech-docs/new-build-system/user-guide/manifest-merger)

Comment (bruno-garcia) [reply sentry-android-core/src/main/AndroidManifest.xml]: I also don't know why, that's why I leaving it out :)
Don't recall code in our 1.x SDK that checks for network status.

For 2.0 we'll need to update the docs and write a migration plan.

**Comment by bruno-garcia** [comment]
@marandaneto I saw the build is broken now.
I should run a local build more often 👿

File: sentry-android-core/src/main/AndroidManifest.xml (hunk 1 above)
Comment (marandaneto) [reply sentry-android-core/src/main/AndroidManifest.xml]: if we would like to have network status, we would need it.
current SDK also does that https://github.com/getsentry/sentry-java/blob/a5b4fd12b65f46a593f0f8317dbfad4aac95097b/sentry-android/src/main/java/io/sentry/android/event/helper/AndroidEventBuilderHelper.java#L546

Comment (bruno-garcia) [reply sentry-android-core/src/main/AndroidManifest.xml]: Fair. If we add then this check we add also the permission then.
Let's discuss whether to do that on a separate PR

**Pull request merged**